src/osd/PrimaryLogPG.cc

```diff
@@ -3801,8 +3801,9 @@ int PrimaryLogPG::do_extent_cmp(OpContext *ctx, OSDOp& osd_op)
     return result;
   }
 
-  if (read_op.outdata.length() != osd_op.indata.length())
-    return -EINVAL;
+  if (read_op.outdata.length() != osd_op.indata.length()) {
+    read_op.outdata.append_zero(osd_op.indata.length() - read_op.outdata.length());
```

Comment:
@dillaman there will be some "HOLE" while there is no data  in the  (offset, length). IMO, that is valid, i only need fill it with zero-data.